torch/distributed/utils.py

```diff
@@ -4,10 +4,40 @@
 from torch.nn.parallel.scatter_gather import (  # type: ignore[attr-defined]
     _is_namedtuple
 )
-from typing import Dict, Any, List
+from typing import Dict, Any, List, Tuple
 
 __all__ = []  # type: ignore[var-annotated]
 
+def _pack_kwargs(*args: Any, **kwargs: Any) -> Tuple[Tuple[str, ...], Tuple[Any, ...]]:
+    """
+    Turn argument list into separate key list and value list (unpack_kwargs does the opposite)
+    Inspiration: https://github.com/facebookresearch/fairscale/blob/eeb6684/fairscale/internal/containers.py#L70
+    Usage::
+
```

Comment:
I wonder if this API design may be slightly confusing due to the inversion of `tuple` ordering.
- From Python, `*args` naturally proceeds `*kwargs`.
- `_pack_kwargs` returns `kwarg_keys` as the first element and `flat_args` as the second element.
- `_unpack_kwargs` takes `kwarg_keys` as the first arg and `flat_args` as the second arg. This nicely matches the return value of `_pack_kwargs`.
- `_unpack_kwargs` returns `args` as the first element and `kwargs` as the second element.

Specifically, I am wondering if we should reorder the `_pack_kwargs` return value to return `flat_args, kwarg_keys` instead. This might be more natural, but this is totally up to you. If you do invert the return value order, then the type annotation and sample docstring above need to be adjusted as well.

Reply:
I see, so we want the intermediate to be consistent with the overall. I think that makes sense.